site/app/libraries/GradingQueue.php

```diff
@@ -209,4 +212,219 @@ public function getGradingCount() {
+
+        foreach ($started as $file) {
+            $path = FileUtils::joinPaths($this->grading_path, $file);
+            $job_file = $this->updateDetailedQueueCount($path, true, $epoch_time, $queue_counts, $capability_queue_counts, $machine_grading_counts, $open_autograding_workers_json, $course_info);
+            $file_segments = explode("/", $file);
+            $machine = substr($file_segments[0], 0, strrpos($file_segments[0], "_", -1));
+            $file_segments = explode("__", $file_segments[1]);
+            $elapsed_time = "";
+            // Calculate the days elapsed
+            if ($job_file["elapsed_time"] > 60 * 60 * 24) {
+                $elapsed_time .= floor($job_file["elapsed_time"] / (60 * 60 * 24)) . " Days ";
+                $job_file["elapsed_time"] = $job_file["elapsed_time"] % (60 * 60 * 24);
+            }
+            // Calculate the hours elapsed
+            $elapsed_time .= str_pad(floor($job_file["elapsed_time"] / (60 * 60)), 2, "0", STR_PAD_LEFT) . ":";
+            $job_file["elapsed_time"] = $job_file["elapsed_time"] % (60 * 60);
+            // Format the string with the minutes and seconds elapsed
+            $elapsed_time .= str_pad(floor($job_file["elapsed_time"] / 60), 2, "0", STR_PAD_LEFT) . ":"
+                . str_pad($job_file["elapsed_time"] % 60, 2, "0", STR_PAD_LEFT);
+            $ongoing_job_info[$machine][] = [
+                "semester" => $file_segments[0],
+                "course" => $file_segments[1],
+                "gradeable_id" => $file_segments[2],
+                "user_id" => $file_segments[3],
+                "regrade" => $job_file["regrade"],
+                "elapsed_time" => $elapsed_time,
+                "error" => $job_file["error"],
+                "stale" => $job_file["stale"]
+            ];
+        }
+
+        // These are files in the grading directory but does not have a GRADING_ file, so grading has not started yet
+        foreach ($not_yet_started as $file) {
+            $path = FileUtils::joinPaths($this->grading_path, $file);
+            $this->updateDetailedQueueCount($path, false, $epoch_time, $queue_counts, $capability_queue_counts, $machine_grading_counts, $open_autograding_workers_json, $course_info);
+        }
+
+        return [
+            "machine_grading_counts" => $machine_grading_counts,
+            "capability_queue_counts" => $capability_queue_counts,
+            "queue_counts" => $queue_counts,
+            "num_autograding_workers" => $workers,
+            "course_info" => $course_info,
+            "ongoing_job_info" => $ongoing_job_info
+        ];
+    }
+
+    // Helper function used to interpret the job files and update the count variables appropriately
+    private function updateDetailedQueueCount(string $queue_or_grading_file, bool $is_grading, int $epoch_time, array &$detailed_queue_counts, array &$capability_queue_counts, array &$machine_grading_counts, array $open_autograding_workers_json, array &$course_info): array {
```

Comment:
Maybe make this function header multiple lines? Also where it gets called too seems long so maybe do the same there too.

Reply:
Is there a way to make the linter happy without using a line for each parameter? I opted for this way unoriginally because I thought it will become too big with one line each for each parameter.

Reply:
I don't know of any way to satisfy the linter without either putting them all on one line or each on its own line.

Reply:
I'm pretty sure you would have to put all arguments on separate line. I think that would still look much better than having 1 very long line.